### ISSUE TITLE
Accessibility : harmful live-region usage

### ISSUE THREAD

**Issue by blokche** [barbajs/barba#681]
I was considering using barba.js for projects but I'm a bit disapointed with the usage of the aria-live region set on the wrapper element. It was a good intention but to me it's more harmful than helpful as it is implemented. 

SPA navigation are often a bit challenging for assistive technologies users but [some patterns have been tested with real users](https://www.gatsbyjs.com/blog/2019-07-11-user-testing-accessible-client-routing/). Live regions can be useful but you don't want the whole document to be read aloud by assistive technologies. Only a page title should be announed in those elements. 

The focus may be moved too, but I guess it's more the responsability of the developer to listen for page transition event and do the right choice for its use case. The documentation may help to build appropriate patterns, without forcing an opiniated behavior.

**Comment by xavierfoucrier** [comment]
Hi @blokche,

Sorry but we are not managing any `aria-live` attribute on the wrapper element.
This is the responsibility of the developer to manager addition work.

Barba JS is here to give you page transition power, [we are aware of SEO concerns](https://barba.js.org/faq/#What-about-SEO-concerns) but the library doesn't manage any additional SEO stuff / attributes, you need to do it yourself.

Thanks for highlighting.

**Comment by blokche** [comment]
Hi. Thanks for your answer. Forgive my insistence but what about this line (186) in  `packages/core/src/core.ts`:
`    this._wrapper.setAttribute('aria-live', 'polite'); // A11y`

**Comment by xavierfoucrier** [comment]
Hi @blokche,

I am not an expert in WAI.
https://www.w3.org/TR/wai-aria/#aria-live

May be @thierrymichel can add further / additional informations to my post.

> Anyway, if you feel that it's not good to use Barba JS library for your project, then feel free to use another one :v:

**Comment by daun** [comment]
Setting aria-live on the wrapper should be avoided since it will read out the entire page on every visit. The ideal solution would be to announce page visits with a [live region](https://www.npmjs.com/package/on-demand-live-region) (e.g. `Navigated to new page: {page title}`) and then focus the new content container. I'd prefer if barba left it up to its users to implement a11y requirements; however you can just as easily remove the `aria-live` attribute in a hook:

```js
barba.hooks.ready(() => {
  const wrapper = barba.dom.getWrapper()
  wrapper.removeAttribute('aria-live')
})
```

**Comment by brettburwell** [comment]
+1 for everything @daun added. 

@xavierfoucrier I believe you may have misunderstood @blokche's original note. @blokche, please correct me if I'm wrong, but I don't believe you were asking for Barba to do _more_, but instead hoping for Barba to actually do less.

Currently, Barba automatically applies a `aria-live="polite"` to the wrapping element. I'm sure the intentions behind this are good, but in practice, it's a bad user experience for anyone with voice over technology on—as it reads out the contents of the _entire wrapper_ every time Barba loads a new page.

@daun's suggestion is a good one. It's a simple way to removes the `aria-live` attribute that Barba adds. But it honestly feels a little silly that it's even necessary. It'd be nice if there was a configuration option that controlled whether or not that attribute is added (or even if the line in question was removed entirely). 

I understand that A11Y stuff can be really challenging and issues don't always have a single, obvious solution. But I think this issue is worth reopening and reconsidering.

**Comment by xavierfoucrier** [comment]
@blokche @daun @brettburwell Hi guys!

Small update, I am now considering this as a bug fix, since it's not a good implementation with most assistive technologies.
I also +1 for what @daun said, announcing page title should be enough.

Thanks everyone for your useful comments 😉

**Comment by brettburwell** [comment]
Hi @xavierfoucrier. 

That's great news! Thanks for taking a second look at this issue and reconsidering.
